### PR TITLE
Update random when assign uses update DB

### PR DESCRIPTION
**Pull request by johnlees**
This is fixed in sketchlib 1.6.5 (https://github.com/johnlees/pp-sketchlib/pull/56), but this automates the refresh of random matches when using `--update-db`

### CODE REVIEW

File: PopPUNK/sketchlib.py [hunk 1]
```diff
@@ -225,35 +225,65 @@ def joinDBs(db1, db2, output):
...
+
+            # Need to close before adding random
+            hdf_join.close()
+            if len(sequence_names) > 2:
+                sys.stderr.write("Updating random match chances")
+                pp_sketchlib.addRandom(join_prefix + ".tmp",
+                                       sequence_names,
+                                       kmer_size,
+                                       not strand_preserved,
+                                       threads)
```
Comment (nickjcroucher) [inline PopPUNK/sketchlib.py:270]: Does this need checking against the number of groups used for the random sequence generation?

Comment (johnlees) [reply PopPUNK/sketchlib.py:270]: I don't think so – it regenerates the whole thing from scratch

Comment (nickjcroucher) [reply PopPUNK/sketchlib.py:270]: I couldn't remember - I know it was 10 groups by default, which caused problems with <10 sequences.

Comment (johnlees) [reply PopPUNK/sketchlib.py:270]: Default clusters is now 3, and we've got the check above here, so think it should be ok

File: PopPUNK/sketchlib.py [hunk 2]
```diff
@@ -225,35 +225,65 @@ def joinDBs(db1, db2, output):
...
+                    threads = update_random["threads"]
+                if "strand_preserved" in update_random:
+                    strand_preserved = update_random["strand_preserved"]
+
+            sequence_names = list(hdf_join['sketches'].keys())
+            kmer_size = hdf_join['sketches/' + sequence_names[0]].attrs['kmers']
+
+            # Need to close before adding random
+            hdf_join.close()
+            if len(sequence_names) > 2:
```
Comment (johnlees) [inline PopPUNK/sketchlib.py:264]: Enough to cluster checked here

**Review by nickjcroucher — APPROVED** [review]
Great!

**Pull request merged**